src/proxy/src/main/java/ee/ria/xroad/proxy/clientproxy/ClientRestMessageHandler.java

```diff
@@ -62,7 +72,7 @@
     MessageProcessorBase createRequestProcessor(String target,
             HttpServletRequest request, HttpServletResponse response,
             OpMonitoringData opMonitoringData) throws Exception {
-
+        requestAcceptType = request.getHeader("Accept");
```

Comment:
getHeader returns only the first header by that name. In case there are multiple Accept headers (possible), the other values are not considered.